crates/rnote-engine/src/engine/import.rs

```diff
@@ -437,14 +465,40 @@ impl Engine {
         self.store.set_selected_keys(&all_strokes, false);
         widget_flags |= self.change_pen_style(PenStyle::Selector);
 
+        // calculate ratio
+        let ratio = match resize {
+            ImageSizeOption::ResizeImage(resize) => {
+                calculate_resize_ratio(resize, content.size().unwrap(), pos)
+            }
+            _ => 1.0f64,
+        };
         let inserted_keys = self.store.insert_stroke_content(content, pos);
+
+        self.store.update_geometry_for_strokes(&inserted_keys);
+        self.store.regenerate_rendering_in_viewport_threaded(
+            self.tasks_tx.clone(),
+            false,
+            self.camera.viewport(),
+            self.camera.image_scale(),
+        );
+
+        self.store
+            .scale_strokes_with_pivot(&inserted_keys, na::Vector2::new(ratio, ratio), pos);
+        self.store.scale_strokes_images_with_pivot(
+            &inserted_keys,
+            na::Vector2::new(ratio, ratio),
+            pos,
+        );
+
+        // re generate view
```

Comment:
This part is the one that's the most fragile.
When resizing of strokes occurs upon insertion (here only when pasting stroke content from an infinite layout to one with a finite width)
- strokes are pasted with no resize (that gives us the keys for the inserted strokes)
- geometry and view is regenerated
- then strokes are rescaled using the keys for the inserted strokes from before
- geometry and view is regenerated

This is one more geometry/view rerender than needed, and it also creates some transient graphical glitch showing phantom strokes that correspond to the original size, before anything gets actually resized..

Reply:
fixed by putting all of that logic inside `insert_stroke_content` directly